compose/service.py

```diff
@@ -395,27 +395,35 @@ def _containers_have_diverged(self, containers):
     def execute_convergence_plan(self,
                                  plan,
                                  do_build=True,
-                                 timeout=DEFAULT_TIMEOUT):
+                                 timeout=DEFAULT_TIMEOUT,
+                                 detached=False):
         (action, containers) = plan
 
         if action == 'create':
             container = self.create_container(do_build=do_build)
+
+            if not detached:
+                container.attach_log_stream()
+
             self.start_container(container)
 
             return [container]
 
         elif action == 'recreate':
+            should_attach_logs = not detached
```

Comment:
This could be moved out of the if/elif/elif block since it's used in all three branches, but super minor, not a blocker